tests/compiler/unary-errors.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-class-assign */
+/* eslint-disable no-global-assign */
+
+class Foo {}
+
+Foo++;
+++Foo;
+// Foo += 1;
+
+Foo--;
+--Foo;
+// Foo -= 1;
+
+Array++;
+++Array;
+// Array += 1;
+
+Array--;
+--Array;
+// Array -= 1;
+
+const a = (Foo++);
+const b = (++Foo);
+// const c = (Foo += 1);
+
+const d = (Foo--);
+const e = (--Foo);
+// const f = (Foo -= 1);
+
+const g = (Array++);
+const h = (++Array);
+// const i = (Array += 1);
+
+const j = (Array--);
+const k = (--Array);
+// const l = (Array -= 1);
+
+ERROR("EOF");  
```

Comment:
I guess all this test may be simplified just to this:
```ts
/* eslint-disable no-class-assign */
/* eslint-disable no-global-assign */

class Foo {}

const a = (Foo--);
const b = (++Foo);

RROR("EOF"); 
```

Reply:
I think more test cases could be useful for the scenario in which someone breaks specific internals and one of the cases fails.

Reply:
`Array` is the same class as Foo and will never have overload ops like `--` or `++`. So I don't see a reason test it. Tests with assignments it seems also covers tests without assignments so it also redundant